src/Symfony/Bundle/FrameworkBundle/Command/UidGenerateCommand.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Uid\AbstractUid;
+use Symfony\Component\Uid\Ulid;
+use Symfony\Component\Uid\Uuid;
+
+class UidGenerateCommand extends Command
+{
+    protected static $defaultName = 'uid:generate';
+
+    private const UUID_1 = '1';
+    private const UUID_3 = '3';
+    private const UUID_4 = '4';
+    private const UUID_5 = '5';
+    private const UUID_6 = '6';
+    private const ULID = 'ulid';
+
+    private static $types = [
+        self::UUID_1,
+        self::UUID_3,
+        self::UUID_4,
+        self::UUID_5,
+        self::UUID_6,
+        self::ULID,
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $typesAsString = implode(', ', self::$types);
+
+        $this
+            ->setDefinition([
+                new InputArgument('type', InputArgument::OPTIONAL, 'The type/version of the generated UID.', null),
```

Comment:
why is this optionnal if you don't allow the null value below ?

Reply:
Indeed, this part was confusing, and the interactive stuff was a bit needless. I've updated the code: now, the `type` argument is require.